lib/fast_jsonapi/object_serializer.rb

```diff
@@ -176,9 +175,23 @@ def set_id(id_name = nil, &block)
       end
 
       def cache_options(cache_options)
-        self.cached = cache_options[:enabled] || false
-        self.cache_length = cache_options[:cache_length] || 5.minutes
-        self.race_condition_ttl = cache_options[:race_condition_ttl] || 5.seconds
+        cache_options = cache_options.dup
+
+        if cache_options.key?(:store)
+          self.cache_store_instance = cache_options.delete(:store)
+        elsif !cache_store_instance
+          warn('DEPRECATION WARNING: `store:` is required, we will default to `Rails.cache`.')
+          self.cache_store_instance =  Rails.cache
+        end
+
+        %i[enabled cache_length].each do |key|
+          if cache_options.key?(key)
```

Comment:
Consider `warn(...) unless cache_options.delete(key).nil?`, this way there's no need to have a nested block and we get rid of the extra call.

Reply:
Yeah, I had this on my mind, too, but then the `.delete` is on the right side and therefore a little bit hidden ... I can change it though, makes the code shorter.

Reply:
Found a way to remove the `.delete` calls, so this is clearer now.